File: apps/mobile/packages/youtube_embedded_player_private_dev/ios/Classes/YouTubeEmbeddedPlayerPlatformViewFactory.swift
#if DEBUG
  import Flutter
  import UIKit

  final class YouTubeEmbeddedPlayerPlatformViewFactory:
    NSObject, FlutterPlatformViewFactory
  {
    static let viewType = "com.moolsocial.app/youtube_embedded_player"

    private let messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
      self.messenger = messenger
      super.init()
    }

    func create(
      withFrame frame: CGRect,
      viewIdentifier viewId: Int64,
      arguments args: Any?
    ) -> FlutterPlatformView {
      YouTubeEmbeddedPlayerPlatformView(
        frame: frame,
        viewId: viewId,
        messenger: messenger
      )
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
      FlutterStandardMessageCodec.sharedInstance()
    }
  }
#endif

File: apps/mobile/packages/youtube_embedded_player_private_dev/android/src/debug/kotlin/com/moolsocial/app/youtube/YouTubeEmbeddedPlayerPlatformViewFactory.kt
package com.moolsocial.app.youtube

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class YouTubeEmbeddedPlayerPlatformViewFactory(
    private val messenger: BinaryMessenger,
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(
        context: Context,
        viewId: Int,
        args: Any?,
    ): PlatformView = YouTubeEmbeddedPlayerPlatformView(
        context = context,
        viewId = viewId,
        messenger = messenger,
    )

    companion object {
        const val VIEW_TYPE = "com.moolsocial.app/youtube_embedded_player"
    }
}
